server/handlers/handlers.go

```diff
@@ -101,6 +102,36 @@ func (ah *ActivityHandler) Start() {
 		}
 	}
 
+	// doStart is the meat of the activity handler worker
+	doStartLastActivityAt := func() {
+		for {
+			timer := time.NewTimer(5 * time.Minute)
+			select {
+			case <-timer.C:
+				ah.lastUpdateAtMap.Range(func(subscriptionID, lastUpdateAt any) bool {
+					if lastUpdateAt.(time.Time).Add(5 * time.Minute).After(time.Now()) {
+						if err := ah.plugin.GetStore().UpdateSubscriptionLastActivityAt(subscriptionID.(string), lastUpdateAt.(time.Time)); err != nil {
+							ah.plugin.GetAPI().LogWarn("Error storing the subscription last activity at", "error", err)
+						}
+					}
+					return true
+				})
+			case <-ah.quit:
+				timer.Stop()
+				// we have received a signal to stop
+				return
+			}
+		}
+	}
+
+	if lastActivityAtMap, err := ah.plugin.GetStore().GetSubscriptionsLastActivityAt(); err != nil {
+		ah.plugin.GetAPI().LogError("Unable to load last activity subscription", "error", err)
+	} else {
+		for subscriptionID, lastActivityAt := range lastActivityAtMap {
+			ah.lastUpdateAtMap.Store(subscriptionID, lastActivityAt)
+		}
+	}
```

Comment:
Can we get away with not populating the map in the first place? I'm thinking about the error case here where something goes wrong and the map starts out empty. The `Range` above will just no-op until the first event, after which the subscription will get updated the very next time the job runs.

Reply:
Something to think about, I think the problem of not having that initialized is about not syncing properly the messages later.

Reply:
I'm assuming when we go to synchronize, we'll want to get the latest values from the database and not rely on the map, as we won't necessarily "see" changes that only get to other plugin instances.

Reply:
Yes, that's true, we can relay on database content, and that should be enough.

Reply:
Discussed offline, but agreed we can just remove this and keep it simpler 🎉